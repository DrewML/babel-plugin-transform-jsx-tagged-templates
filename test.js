const babel = require('babel-core');
const plugin = require('.');
const fs = require('fs');

const t = input => {
    const { code, ast } = babel.transform(input, {
        plugins: ['syntax-jsx', plugin]
    });
    return code;
};

test('single self-closing', () => {
    expect(t('<div />')).toMatchSnapshot();
});

test('nested self-closing', () => {
    expect(t('<div><span /></div>')).toMatchSnapshot();
});

test('JSXText child', () => {
    expect(t('<div>foo</div>')).toMatchSnapshot();
});

test('dynamic attr', () => {
    expect(t('<div id={1}></div>')).toMatchSnapshot();
});

test('2 props, 1 dynamic 1 static, dynamic first', () => {
    expect(t('<div id={1} b="a"></div>')).toMatchSnapshot();
});

test.skip('2 props, 1 dynamic 1 static, static first', () => {
    expect(t('<div b="a" id={1}></div>')).toMatchSnapshot();
});
