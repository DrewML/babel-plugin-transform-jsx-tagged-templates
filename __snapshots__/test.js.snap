// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`2 props, 1 dynamic 1 static, dynamic first 1`] = `"html\`<div id=\${1} b=\\"a\\"></div>\`;"`;

exports[`JSXText child 1`] = `"html\`<div>foo</div>\`;"`;

exports[`dynamic attr 1`] = `"html\`<div id=\${1}></div>\`;"`;

exports[`nested self-closing 1`] = `"html\`<div><span></span></div>\`;"`;

exports[`single self-closing 1`] = `"html\`<div></div>\`;"`;
